ci/linux-debian.Dockerfile

```diff
@@ -23,6 +25,29 @@ RUN apt-get update && apt-get install --no-install-recommends -y \
         sagemath
 
 WORKDIR /root
+
+# Install gcc snapshot
+RUN wget --progress=dot:giga --content-disposition https://kayari.org/gcc-latest/gcc-latest.deb && \
+    dpkg -i gcc-latest_*.deb && \
+    ln -s /opt/gcc-latest/bin/gcc /usr/bin/gcc-snapshot
```

Comment:
This installs `gcc version 13.0.1 20230416`. There are newer gcc snapshot sources available from the [official mirrors](https://gcc.gnu.org/mirrors.html), e.g., [gcc-14-20230625](https://ftp.gwdg.de/pub/misc/gcc/snapshots/LATEST-14/). I suppose you didn't pick this because we don't want to get into the business of building the compiler (for now)?

Reply:
Hm, yeah, it seems that this build is already ~2.5 months old.

> I suppose you didn't pick this because we don't want to get into the business of building the compiler (for now)?

Actually not.

The gcc team publishes development snapshots *in source form* (what you linked to).  https://jwakely.github.io/pkg-gcc-latest/ picks these up and provides binary packages (what we install here). I wonder if there's a regular cadence. Asked here: https://github.com/jwakely/pkg-gcc-latest/issues/14

Reply:
It's supposed to be every week.

Reply:
[GCC 14 snapshot binaries are available now.](https://github.com/jwakely/pkg-gcc-latest/issues/14#issuecomment-1618808302)